extractor/filesystem/language/rust/cargolock/extractor.go

```diff
@@ -0,0 +1,90 @@
+// Package cargolock extracts Cargo.lock files for rust projects
+package cargolock
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+)
+
+type cargoLockPackage struct {
+	Name    string `toml:"name"`
+	Version string `toml:"version"`
+}
+
+type cargoLockFile struct {
+	Version  int                `toml:"version"`
+	Packages []cargoLockPackage `toml:"package"`
+}
+
+const cargoEcosystem string = "crates.io"
+
+// Extractor extracts crates.io packages from Cargo.lock files.
+type Extractor struct{}
+
+// Name of the extractor
+func (e Extractor) Name() string { return "rust/cargolock" }
+
+// Version of the extractor
+func (e Extractor) Version() int { return 0 }
+
+// FileRequired returns true if the specified file matches Cargo lockfile patterns.
+func (e Extractor) FileRequired(path string, _ fs.FileInfo) bool {
+	// TODO: File size check?
```

Comment:
Add bug references with TODOs.

We only do file size checks on a select few extractors that internal teams wanted to run in resource constrained environments so it should be fine to just remove the TODO from here since this will mostly be used in code scanning environments.

Reply:
Removed